Issue #22 Use drupal/recommended-project

giteshk opened the pull request:
Update project to use Composer template drupal/recommended-project. Fixes #22

giteshk committed:
Update project to use Composer template drupal/recommended-project

giteshk committed:
Added a test for packagist

giteshk committed:
Fixed formatting

giteshk commented:
This is what the welcome message looks like:

![Screen Shot 2020-03-20 at 11 32 13 AM](https://user-images.githubusercontent.com/6208023/77179230-8f504180-6a9e-11ea-8154-d32ee3589780.png)


`
Writing lock file
Generating autoload files
PHP CodeSniffer Config installed_paths set to ../../drupal/coder/coder_sniffer

  Congratulations, you’ve installed the Apigee Developer Portal Drupal codebase
  from the apigee/devportal-kickstart-project template!


Next steps:
  * Install the site: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/use-kickstart-with-apigee-edge-for-private-cloud
  * Read the guide: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart
  * Report issues : https://github.com/apigee/apigee-devportal-kickstart-drupal/issues
  * Homepage: https://www.drupal.org/project/apigee_devportal_kickstart
  * Support:
    * docs: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart
`

giteshk commented:
Let me finish some more testing before merging this.

giteshk committed:
Fixed the missing package + PHP version bumped up to 7.2

giteshk commented:
Finished testing. Ready for review

arlina-espinoza commented on composer.json:
```diff
@@ -3,47 +3,73 @@
+    "drupal/core-recommended": "^8.8",
     "apigee/apigee_devportal_kickstart": "^1.0.0",
     "drupal/apigee_m10n": "^1.0.0",
     "drupal/commerce": "^2.13",
```
The latest commerce requires commerce >=8.x-2.16, so we can bump this up.
```suggestion
    "drupal/commerce": "^2.16",
```

arlina-espinoza commented on composer.json:
```diff
@@ -3,47 +3,73 @@
+      "web/modules/custom/{$name}": ["type:drupal-custom-module"],
+      "web/themes/custom/{$name}": ["type:drupal-custom-theme"]
     },
-    "enable-patching": true
```
We need to keep this line to apply patches from dependencies - `apigee/apigee_devportal_kickstart` has at least one patch defined in it's composer.json file.

giteshk replied:
added this back. Tested this.

arlina-espinoza reviewed (changes requested):
Just a couple of comments, but it looks great. So nice to see the post-install message with helpful links! :heart:

cnovak commented on composer.json:
```diff
@@ -3,47 +3,73 @@
+        "<bg=blue;fg=white>                                                                                 </>",
+        "",
+        "<bg=yellow;fg=black>Next steps</>:",
+        "  * Install the site: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/use-kickstart-with-apigee-edge-for-private-cloud",
```
I put info on all ways to install on this page to be more generic: 
```suggestion
        "  * Install the site: https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/get-started-with-kickstart",
```

cnovak commented on composer.json:
```diff
@@ -3,47 +3,73 @@
     }
   ],
-  "config": {
-    "process-timeout": 0
```
This line makes sure the quick start command does not timeout this is what happens:
```Drupal development server started: <http://127.0.0.1:8888>
This server is not meant for production use.
One time login url: <http://127.0.0.1:8888/user/reset/1/1584984687/CzeQ3WiAoU2qnUwniMEissUExPpt32sHXFYpsx4pvXs/login>
Press Ctrl-C to quit the Drupal development server.
  [Symfony\Component\Process\Exception\ProcessTimedOutException]
  The process "php web/core/scripts/drupal quick-start apigee_devportal_kickstart" exceeded the timeout of 300 seconds.
```

The server will also not start a second time, due to `Could not delete /devportal-kickstart-project-composer/web/sites/default/default.services.yml`, see https://www.drupal.org/project/drupal/issues/3091285.  Should add a fix to the readme that you can `sudo chmod -R 777  web` to run again.

giteshk replied:
I verified this fix. Added this back.

cnovak reviewed (changes requested):
Notes inline, and:
The README needs to be modified to remove `  --no-interaction`  from `composer create-project` command, and these locations also need that fix:
- Project home page for Apigee Kickstart on Drupal.org
- https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/get-started-with-kickstart

giteshk committed:
Update composer.json

Co-Authored-By: Chris Novak <chrisnovak@google.com>

giteshk committed:
Update composer.json

Co-Authored-By: Arlina Espinoza Rhoton <arlina.espinoza@gmail.com>

giteshk committed:
1. Re-Added the ability for the dependencies to apply patches
2. Added the process-timeout:0 to enable quickstart to run longer then 5 minutes.

giteshk committed:
Readme file updated

giteshk commented:
> Notes inline, and:
> The README needs to be modified to remove ` --no-interaction` from `composer create-project` command, and these locations also need that fix:
> 
> * Project home page for Apigee Kickstart on Drupal.org
> * https://www.drupal.org/docs/8/modules/apigee-developer-portal-kickstart/get-started-with-kickstart


@cnovak I applied this change to this commit and updated the docs on those listed links

cnovak reviewed (approved):
LGTM

arlina-espinoza reviewed (approved):
:+1:
